src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute.Fluent/VirtualMachineScaleSetExtensionImpl.cs

```diff
@@ -63,7 +60,16 @@ public bool AutoUpgradeMinorVersionEnabled()
         ///GENMHASH:E8B034BE63B3FB3349E5BCFC76224AF8:CA9E3FB93CD214E58089AA8C2C20B7A3
         public IReadOnlyDictionary<string, object> PublicSettings()
         {
-            return this.publicSettings as IReadOnlyDictionary<string, object>;
+            if (this.Inner.Settings == null)
+            {
+                return new Dictionary<string, object>();
+            }
+            if (this.Inner.Settings is JObject)
+            {
+                var jObject = this.Inner.Settings as JObject;
```

Comment:
>as JObject [](start = 50, length = 10)

Please change "as" cast to explicit conversion. there are to location in this method.